test/envtest/run.go

```diff
@@ -208,3 +229,14 @@ func RunManager(
 
 	return cfg, logs
 }
+
+// WaitForManagerStart waits for the manager to start.
+func WaitForManagerStart(t *testing.T, logs observerLogs) {
+	t.Log("Waiting for manager to start...")
```

Comment:
- I'd suggest to call the second argument a bit different for the reason that we're getting the logs each iteration via `logs.All()`: these are not "logs" per se but rather a means to get the current logs from the container. Those being "logs" would imply that `logs.All()` always return the same logs. WDYT?
- Can we add a godoc comment describing what are the conditions under which the manager is considered started?

```suggestion
func WaitForManagerStart(t *testing.T, logs observerLogs) {
	t.Helper()
	t.Log("Waiting for manager to start...")
```

Reply:
Added a godoc explaining what it does and why: https://github.com/Kong/kubernetes-ingress-controller/pull/5605/commits/4a88ffc9295bcb83d60f14bc9ac8f7dc1474e888 👍